src/main/java/seedu/duke/manager/OrderManager.java

```diff
@@ -1,46 +1,121 @@
 package seedu.duke.manager;
 
+import java.util.Arrays;
+import java.util.List;
 import seedu.duke.entities.Dish;
 import seedu.duke.entities.Order;
+import seedu.duke.loggers.MainLogger;
 
 import java.util.ArrayList;
 
-
-public class OrderManager {
-    private final ArrayList<Order> orders;
+/**
+ * OrderManager.
+ */
+public class OrderManager extends Manager {
+    private static OrderManager singleton = null;
+    private List<Order> orders;
     private final DishManager dishManager = DishManager.getInstance();
-    private final ArrayList<Dish> dishes;
 
-    public OrderManager() {
-        this.orders = new ArrayList<Order>();
-        this.dishes = dishManager.getDishes();
+    private static final String ORDER_STORAGE_FILE = "order.dat";
+
+    private OrderManager() {
+        super(ORDER_STORAGE_FILE);
+        try {
+            this.loadData();
+        } catch (Exception e) {
+            MainLogger.logWarning(this, e.toString());
+            MainLogger.logWarning(this, "Start with an empty order");
+            orders.clear();
+        }
+    }
+
+    public static OrderManager getInstance() {
+        if (singleton != null) {
+            return singleton;
+        }
+        singleton = new OrderManager();
+        return singleton;
+    }
+
+    public static void resetInstance() {
+        singleton = null;
     }
 
+    @Override
+    protected void loadData() throws Exception {
+        this.orders = new ArrayList<>();
+        ArrayList<?> list = (ArrayList<?>) this.load();
+        for (Object object : list) {
+            this.addOrder((Order) object);
+        }
+    }
+
+    @Override
+    public void saveData() throws Exception {
+        this.save(this.orders);
+    }
+
+    /**
+     * Add the created order to the order list.
+     *
+     * @param order the created order.
+     */
     public void addOrder(Order order) {
         this.orders.add(order);
     }
 
-    public int addDishToOrder(Object inputObj) {
-        Order order = null;
-        if (inputObj instanceof Integer && ((Integer) inputObj <= this.dishes.size())) {
-            int dishIdx = (Integer) inputObj;
-            Dish dish = this.dishes.get(dishIdx);
-            order.addDishToOrder(dish);
-        } else {
-            System.out.println("Please enter a valid input: number of dish in the menu.");
+
+    /**
+     * Add dish by index to the order.
+     *
+     * @param dishIdx the index of the dish.
+     * @return size of the order list.
+     */
+    public void addDishToOrder(int dishIdx, int orderIdx) {
+        ArrayList<Dish> dishes = dishManager.getDishes();
+        Dish dish = dishes.get(dishIdx);
```

Comment:
then here, you can rewrite it such that it will pass in the orderIndex and the Dish object as the parameters. you can get the dish object in the order controller, rather than inside the order manager.

Reply:
if not we have coupling between managers, which was not the intended design.